content/ros/ROS Installation/ros_catkin_installer.sh

```diff
@@ -1,159 +1,284 @@
 
-# ROS installation and catkin initialization script
+#####################################################################################
+##                                                                                 ##
+##  ROS Installation script                                                        ##
+##                                                                                 ##
+##  This script acts as a little tool to install all necessary ROS packages        ##
+##  needed for the Autonomous Mobile Robots course at HBRS.                        ##
+##  It can also set up a catkin workspace and add ROS and catkin to your basrc.    ##
+##                                                                                 ##
+##  Note: This has been tested on a fresh Ubuntu 16.04 and with ROS Kinetic Kame   ##
+##  Tim Metzler 09/2018                                                            ##
+##                                                                                 ## 
+#####################################################################################
+
+# ROS packages
+# This is the ROS version that is used (i.e. indigo, kinetic, etc)
+ROS_DISTRO=kinetic
+
+# These are the core ROS packages that are installed.
+# Each package in the array will be converted to ros-$ROS_DISTRO-pkg_name (ros-kinetic-desktop)
+ROS_PKGS=(desktop stage smach-viewer map-server)
+
+# These are the python packages needed.
+# Each package in the array will be converted to python-pkg_name (e.g. python-rosinstall)
+PYTHON_PKGS=(catkin-tools rosinstall rosinstall-generator wstool pyside)
+
+# Additional packages to be installed
+OTHER_PKGS=(build-essential gnuplot)
+
+# Directory in which the script is executed
+DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
+
+function is_installed {
+  # Check if the package given as the first argument $1 is installed
+  # For this the dpkg-query output is searched for the keyphrase 'install ok installed'
+  PKG_INSTALLED=$(dpkg-query -W --showformat='${Status}' $1 2>/dev/null|grep "install ok installed")
+  if [ "$PKG_INSTALLED" == "install ok installed" ];
+  then
+    return 0
+  else
+    return 1
+  fi
+}
+
+function install_pkg {
+  # Install the package given as the first argument $1
+  # Check if it is already present and check after installation for success
+  # If the installation is unsuccessful exit this script
+  echo ""
+  echo "Installing $1..."
+  # Check if package is already installed
+  if is_installed $1;
+  then
+    echo ""
+    echo "$1 is already installed"
+    return 0
+  else
+    # Install the package
+    sudo apt-get -y install $1
+    # Check if package was successfully installed
+    if is_installed $1;
+    then
+      echo ""
+      echo "Successfully installed package $1"
+      return 0
+    else
+      echo ""
+      echo "ERROR: Package $1 could not be installed!"
+      echo "Installation aborted!"
+      exit 1
+    fi
+  fi
+}
 
 function update_keys {
```

Comment:
If we aim to reduce the number of times sudo is required, in principle `update_keys` and `ros_install` can be a single function.